docs/gafaelfawr.rst

```diff
@@ -0,0 +1,36 @@
+###################################
+Requiring Gafaelfawr authentication
+###################################
+
+Safir provides two FastAPI dependencies intended for applications designed to run behind a Kubernetes ingress configured to authenticate requests with `Gafaelfawr`_.
+
+Getting the authenticated user
+==============================
+
+Gafaelfawr sets the ``X-Auth-Request-User`` HTTP header to the username of the user authenticated with a Gafaelfawr token.
+To get that username, use the `~safir.dependencies.gafaelfawr.auth_dependency` FastAPI dependency, as follows:
+
+.. code-block:: python
+
+   from fastapi import Depends
+
+   from safir.dependencies.gafaelfawr import auth_dependency
+
+
+   @app.get("/route")
+   async def get_rounte(user: str = Depends(auth_dependench)) -> Dict[str, str]:
+       # Route implementation using user.
+       return {"some": "data"}
+
+If the request does not have the Gafaelfawr header set, it will be rejected by FastAPI with a 422 status code.
```

Comment:
Should something be said about testing an app that uses this dependency? Do we have a recommended test client that bakes in this header? Or remind the user that their test client should include this header?

Reply:
Good idea. I added a new section.